exercises/diffie-hellman/src/test/java/DiffieHellmanTest.java

```diff
@@ -0,0 +1,95 @@
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertEquals;
+
+import java.math.BigInteger;
+import java.util.*;
+
+public class DiffieHellmanTest {
+
+    private DiffieHellman diffieHellman;
+
+    @Before
+    public void setUp() {
+        diffieHellman = new DiffieHellman();
+    }
+
+    @Test
+    public void testPrivateKeyInRange() {
+        BigInteger primeP = BigInteger.valueOf(23);
+        List<BigInteger> privateKeys = new ArrayList<>();
+        for (int i = 0; i < 10; i++) privateKeys.add(diffieHellman.privateKey(primeP));
+
+        for (BigInteger privateKey : privateKeys) {
+            assertTrue((privateKey.compareTo(BigInteger.ONE) >= 0) && (privateKey.compareTo(primeP) < 0));
+        }
+    }
+
+    // Note: due to the nature of randomness, there is always a chance that this test fails
+    // Be sure to check the actual generated values
+    @Ignore("Remove to run test")
+    @Test
+    public void testPrivateKeyRandomlyGenerated() {
+        BigInteger primeP = BigInteger.valueOf(7919);
+        Set<BigInteger> privateKeys = new HashSet<>();
+        for (int i = 0; i < 5; i++) privateKeys.add(diffieHellman.privateKey(primeP));
+
+        assertEquals(5, privateKeys.size());
+    }
+
+    @Ignore("Remove to run test")
+    @Test
+    public void testPublicKeyCorrectlyCalculated() {
+        BigInteger primeP = BigInteger.valueOf(23);
+        BigInteger primeG = BigInteger.valueOf(5);
```

Comment:
Similarly to my earlier comment, I would name these something like `firstPrime` and `secondPrime` unless the letters `P` and `G` are significant? 🙂

Reply:
Maybe ```primeA``` and ```primeB``` would be better names. Like ```publicKeyA```, ```publicKeyB```, ```privateKeyA```, ```privateKeyB```, ```secretA``` and ```secretB``` on the last test 🙂